testsuite/MDAnalysisTests/core/test_topologyattrs.py

```diff
@@ -543,3 +543,19 @@ def test_segment_record_types(self, rectype_uni):
         assert not rt[0].dtype == bool
         assert (rt[0] == 'ATOM').all()
         assert (rt[1] == 'HETATM').all()
+
+
+def test_static_typing():
```

Comment:
What does the error look like if the type of the input is wrong? Is it understandable by the user?

Reply:
@jbarnoud 

```python
In [3]: u.add_TopologyAttr('charges', [3.5, 'a', 4.6])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-e440b376cd93> in <module>()
----> 1 u.add_TopologyAttr('charges', [3.5, 'a', 4.6])

~/miniconda3/envs/mda/lib/python3.6/site-packages/MDAnalysis/core/universe.py in add_TopologyAttr(self, topologyattr, values)
    829                     n_residues=self._topology.n_residues,
    830                     n_segments=self._topology.n_segments,
--> 831                     values=values)
    832         self._topology.add_TopologyAttr(topologyattr)
    833         self._process_attr(topologyattr)

~/miniconda3/envs/mda/lib/python3.6/site-packages/MDAnalysis/core/topologyattrs.py in from_blank(cls, n_atoms, n_residues, n_segments, values)
    253         elif cls.dtype is not None:
    254             # if supplied starting values and statically typed
--> 255             values = np.asarray(values, dtype=cls.dtype)
    256         return cls(values)
    257

~/miniconda3/envs/mda/lib/python3.6/site-packages/numpy/core/numeric.py in asarray(a, dtype, order)
    490
    491     """
--> 492     return array(a, dtype, copy=False, order=order)
    493
    494

ValueError: could not convert string to float: 'a'
```